### DISCUSSION TITLE
Create a basic CRUD admin panel for Shield

### DISCUSSION THREAD

**Discussion by TwixtedChaox** [codeigniter4/shield#531] [Ideas]
I know this has probably been stated a thousand times.
Beginners to CI4 and Shield have no clue where to start with such an undertaking.

We should develop a basic admin area specifically for shield management. Implementing access to the permissions seems easy enough on my end but I've had a lot of people ask how they can manage the users.

Such a thing would help the "reverse" engineers out there better understand the entire shield project and implement shield into projects even faster

**Comment by MGatner** [comment]
I an admin panel is a great idea, but IMO it does not belong in this library. It's already very difficult to leave frontend UI components "open" enough for people to make them fit into any project. An admin panel would lock in a design, CSS & JS frameworks, etc.

There's plenty of space for someone to make one, and I could imagine some sort of "extensions" or "featured libraries" section in the docs where we highlight developer works.

**Reply by TwixtedChaox** [reply]
I agree with this. Seeing some of the other comments.... I never saw Bonfire until now! If that were a highlighted project in the docs, it would give a lot of budding developers a huge insight.

**Comment by lonnieezell** [comment]
Yes, a third-party solution would be best. If we were to make it ourselves I would want it to be a separate package that you can then choose which CSS solution you're using, etc. 

However, if you need something to reverse engineer, you can take a look at [Bonfire](https://github.com/lonnieezell/Bonfire2), an admin area that I've put together.

**Reply by TwixtedChaox** [reply]
This is amazing. Up until you mentioned it, Bonfire didnt come up in any searches for "Shield Admin" or any variation of the search.
They should absolutely highlight this in the docs as a showcased work.

**Reply by TwixtedChaox** [reply]
You absolutely should post this in the "Show and tell" section. I looked there first for unique ideas and this would have definitely popped out!

**Comment by jozefrebjak** [comment]
I personally like how it's done now. We have such freedom. Bonfire 2 is a perfect source on how to implement user management. I learned a lot from Bonfire 2, but it doesn't fit in my projects as a complete package. I want to use my own assets and implement my own features, so maybe suggesting somewhere that Bonfire 2 is a place to learn how to use shield will be enough. Rather than build the whole Frontend, we can write some commands for basic user management.

**Reply by TwixtedChaox** [reply]
I agree here. My biggest concern was that up until now..... I hadnt even heard of Bonfire... no google searches came up with it as an option even. If they added this as a showcased work so that its easier to find I believe it would help even  more developers to understand the intricate details of how shield can be applied.

I also agree that while it is beautiful to see it all together, i dont believe it will work for most projects out there.
